src/components/DatBoolean.vue

```diff
@@ -1,23 +1,38 @@
 <script setup lang="ts">
+import { computed } from "vue";
+
 interface Props {
   label?: string;
+  disabled?: boolean;
 }
 
-withDefaults(defineProps<Props>(), {
+const props = withDefaults(defineProps<Props>(), {
   label: "",
+  disabled: false,
+});
+const watchDisabled = computed(() => {
+  return props.disabled;
 });
 const checked = defineModel({ type: Boolean });
 const handleChange = () => {
+  if (watchDisabled.value) {
+    return;
+  }
```

Comment:
We can keep it as a safety stop. But I think `disabled` on the `input` is enough and this method wouldn't be called.